examples/file-upload.html

```diff
@@ -0,0 +1,181 @@
+<title>snuggsi ツ - File Upload</title>
+
+<meta charset=UTF-8>
+<meta name=viewport content=width=device-width,initial-scale=1>
+
+<link rel=stylesheet href=http://www.devpunks.com/typography.css>
+
+<script nomodule    src=https://unpkg.com/snuggsi/snuggsi.js></script>
+<script nomodule    src=https://unpkg.com/snuggsi/examples/webcomponents-hi-ce.js></script>
+
+<hgroup>
+  <h1>snuggsi ツ</h1>
+  <h2>File Upload</h2>
+</hgroup>
+
+<file-upload>
+
+  <input multiple id=files name=files[] type=file>
+  <label for=files ondragover="event.preventDefault()" ondragenter=ondragstart ondragleave=ondragend ondrop=ondragdrop>
+    Click to upload file...
+  </label>
+
+  <table style="width:300px">
+    <thead style="text-align: left">
+      <tr>
+        <th width="1%"></th>
+        <th>Name</th>
+        <th width="10%">Size</th>
+        <th width="1%"></th>
+      </tr>
+    </thead>
+    <tbody>
+      <template name=files>
+        <tr>
+
+          <td>
+            <img src={icon_url} alt={type} hidden={hidden} style="width: 1em; height: 1em">
+          </td>
+          <td>
+            {name}
+          </td>
+          <td>
+            {size}
+          </td>
+          <td style="cursor:pointer">
+            <a onclick=onremove data-index={#}>&times;</a>
+          </td>
+        </tr>
+      </template>
+    </tbody>
+  </table>
+
+  <a onclick=onclear>Clear</a>
+
+<style>
+
+file-upload input[type=file] {
+  opacity: 0;
+  z-index: -1;
+  width: 0.1px;
+  height: 0.1px;
+  overflow: hidden;
+  position: absolute
+}
+file-upload input[type=file] + label {
+  display: block;
+  padding: 2em 0;
+  margin: 1em 20%;
+  cursor: pointer;
+  font-size: 1.25em;
+  text-align: center;
+  color: rgba(20,20,20,0.6);
+  background-color: rgba(240,240,240, 0.8)
+}
+
+file-upload input[type=file] + label:hover
+  { background-color: rgba(240,240,240, 1.0) }
+
+</style>
+
+<script>
+
+Element `file-upload`
+
+(class extends HTMLElement {
+
+  initialize ()
+    { this.context.files = [] }
+
+
+  static onchange (event) {
+    if (!!! event.target.files) return
+
+    this.add_files(event.target.files)
+
+    this.render ()
+  }
+
+  static onremove (event) {
+    this.context.files.splice(event.target.dataset.index, 1)
+
+    this.render ()
+  }
+
+  static onclear (event) {
+    this.context.files = []
+
+    this.render ()
+  }
+
+  static ondragstart (event) {
+    event.target.innerHTML = 'Drop here!'
+
+    event.preventDefault ()
```

Comment:
@brandondees noticing that `event.preventDefault ()` MAY BE MUTUALLY EXCLUSIVE with `this.render ()` which would be a great thing. Perhaps when someone writes `event.preventDefault ()` we don't render otherwise will render component "automagically" thus removing the need to explicitly call `this.render ()` from event handlers. 

Thoughts @VicenteRD? The beauty is people coming from jQuery and vanilla JS have been familiar with this pattern for over a decade.

To be clear this would be a subsequent PR not in this one. Wanna get this one in asap.